libr/core/core.c

```diff
@@ -1211,7 +1211,8 @@ R_API int r_core_serve(RCore *core, RIODesc *file) {
 			return -1;
 		if (c == NULL) {
 			eprintf ("rap: cannot accept\n");
-			r_socket_close (c);
+			/*r_socket_close (c);*/
+			r_socket_free (c);
```

Comment:
freeing a socket should close it, but im not sure if this will produce a regression or not. but the correct or expected functionality should be this one.